service/dap/server.go

```diff
@@ -214,43 +214,18 @@ func NewServer(config *service.Config) *Server {
 
 // If user-specified options are provided via Launch/AttachRequest,
 // we override the defaults for optional args.
-func (s *Server) setLaunchAttachArgs(request dap.LaunchAttachRequest) error {
-	stop, ok := request.GetArguments()["stopOnEntry"].(bool)
-	if ok {
-		s.args.stopOnEntry = stop
-	}
-	depth, ok := request.GetArguments()["stackTraceDepth"].(float64)
-	if ok && depth > 0 {
-		s.args.stackTraceDepth = int(depth)
-	}
-	globals, ok := request.GetArguments()["showGlobalVariables"].(bool)
-	if ok {
-		s.args.showGlobalVariables = globals
-	}
-	paths, ok := request.GetArguments()["substitutePath"]
-	if ok {
-		typeMismatchError := fmt.Errorf("'substitutePath' attribute '%v' in debug configuration is not a []{'from': string, 'to': string}", paths)
-		pathsParsed, ok := paths.([]interface{})
-		if !ok {
-			return typeMismatchError
-		}
-		clientToServer := make([][2]string, 0, len(pathsParsed))
-		serverToClient := make([][2]string, 0, len(pathsParsed))
-		for _, arg := range pathsParsed {
-			pathMapping, ok := arg.(map[string]interface{})
-			if !ok {
-				return typeMismatchError
-			}
-			from, ok := pathMapping["from"].(string)
-			if !ok {
-				return typeMismatchError
-			}
-			to, ok := pathMapping["to"].(string)
-			if !ok {
-				return typeMismatchError
-			}
-			clientToServer = append(clientToServer, [2]string{from, to})
-			serverToClient = append(serverToClient, [2]string{to, from})
+func (s *Server) setLaunchAttachArgs(args LaunchAttachCommonConfig) error {
+	s.args.stopOnEntry = args.StopOnEntry
+	if depth := args.StackTraceDepth; depth > 0 {
```

Comment:
Is this logic needed now that you populate args with defaults too?

Reply:
yes, it is needed in case the client sends 0 or a negative value.